ZEP-0008 Introduce `zarf dev inspect` and change `zarf package inspect` to a parent command (#9)
zarf-dev/proposals: 0008-rework-inspect/README.md

  - [Risks and Mitigations](#risks-and-mitigations)
- [Design Details](#design-details)
  - [Test Plan](#test-plan)
      - [Prerequisite testing updates](#prerequisite-testing-updates)
      - [Unit tests](#unit-tests)
      - [e2e tests](#e2e-tests)
  - [Graduation Criteria](#graduation-criteria)
  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
  - [Version Skew Strategy](#version-skew-strategy)
- [Implementation History](#implementation-history)
- [Drawbacks](#drawbacks)
- [Alternatives](#alternatives)
<!-- /toc -->

## Summary

<!--
This section is key for creating high-quality, user-focused documentation
like release notes or a roadmap. You should gather this info before
implementation starts to keep the focus on development, not writing. ZEP
editors should ensure the `Summary` is clear and useful for a broad audience.

A good summary should be at least a paragraph long.

Follow the [documentation style guide] for this section and the rest of the ZEP.
Keep line lengths reasonable to make it easier for reviewers to provide
feedback and reduce unnecessary changes.

[documentation style guide]: https://docs.zarf.dev/contribute/style-guide/
-->

> The `zarf package inspect` command will become a parent command. It will have the five subcommands: `definition`, `sbom`, `images`, `manifests`, and `values-files`. The `package inspect` commands will accept an already built package, either local or OCI. `definition` and `images` will also accept a package in the cluster. The `zarf dev inspect` parent command will be introduced. It will have three subcommands: `definition`, `sbom` and `values-files`. These commands will accept a directory containing a `zarf.yaml` file.

Author: Gotcha, yeah summary should be aimed at the clear value end users are receiving from this change

## Motivation

<!--
This section is for explicitly listing the motivation, goals, and non-goals of
this ZEP.  Describe why the change is important and the benefits to users. You
can also optionally include links to [experience reports], [community slacks],
or other references to show the community's interest in the ZEP.

[experience reports]: https://go.dev/wiki/ExperienceReports
[openssf slack]: https://openssf.slack.com/archives/C07AKUMBDMJ
[kubernetes slack]: https://kubernetes.slack.com/archives/C03B6BJAUJ3
-->

Users need an easier way to view their package definition after it's rendered by Zarf, but before `zarf package create`. A rendered package definition has had templating, imports, and flavors applied. The only path to view the rendered package definition is running `zarf package create` and viewing the printed yaml before the (y/n) prompt. Having a separate command, `zarf dev inspect definition`, improves the UX by providing users with an easier way to view the rendered package definition. It also opens the possibility of allowing `zarf package create` to proceed without requiring user confirmation.

> Viewing manifests and values files after Zarf variable templating would be useful for both creators and deployers. Catching a mistake in templating early can reduce cycle time. A Helm template is almost instant, whereas create + deploy could take several minutes to hours.

Reviewer: ```suggestion
Viewing manifests and values files after Zarf variable templating would be useful for both creators and deployers. Catching a mistake in templating early can save time. A Helm template is almost instant, whereas create + deploy could take several minutes to hours.
```

A user can achieve a similar effect to `zarf package inspect manifests` by decompressing their package and running `helm template` on their chart. Not only is this a poor UX, but the `helm template` may fail depending on where Zarf variable templating is used within the chart.

Features relating to these problems have been highly requested in recent months:
- Request in Kubernetes Slack - https://kubernetes.slack.com/archives/C03B6BJAUJ3/p1730229638367829
- An issue has been created for this - https://github.com/zarf-dev/zarf/issues/2631
- Defense Unicorns, an organization that relies heavily on Zarf for their deployments, has received requests to view manifests in a feedback session with their partners.

### Goals

<!--
List the specific goals of the ZEP. What is it trying to achieve? How will we
know that this has succeeded?
-->

- View manifests or values files after Zarf variable templating and Helm templating have been applied.

> - View rendered package definition before create

Reviewer: ```suggestion
- View rendered package definition before creation 
```

and make progress.
-->

- Accept a cluster source of the package for new functionality. Cluster source will continue to work for `zarf package inspect definition` and `zarf package inspect images`, but not for `zarf package inspect sbom`, `zarf package inspect manifests`, or `zarf package inspect values-files`
- Print out multiple types of inspects in the same command. A user will not be able to print out the package definition and manifests in the same command. 
- Support directly opening the SBOM in browser

## Proposal

<!--
This is where you explain the specifics of the proposal. Provide enough detail
for reviewers to clearly understand what you're proposing, but avoid including
too many specifics like API designs or implementation details. Focus on the
desired outcome and how success will be measured. The "Design Details" section
below is for the real nitty-gritty.
-->

These commands will not have a confirm flag and will not prompt for optional components, package templates, or package variables. Users will be able to specify these values using flags, when applicable. None of these commands will run any Zarf actions.

### zarf package inspect

> `zarf package inspect` will be deprecated and replaced by the five commands below

Reviewer: ```suggestion
`zarf package inspect` will be deprecated and replaced by the five commands below.
```

#### zarf package inspect definition

> This will mirror behavior of the current `zarf package inspect <package>` command

Reviewer: Nit, I'd avoid printing help or limit code fragments in enhancements. Only where strictly necessary (eg. API types) it seems reasonable. All other instances are better expressed in plain English. This ensures that implementation and proposal don't drift.

Author: Fair point, there would definitely be drift. I will edit to remove code blocks
